tests/frontend/test_configuration.py

```diff
@@ -46,107 +46,371 @@
+        config = conf.create_config_object()
+        assert config["api"][specific_key] != "PlaceHolder"
+
+        conf.update_with_other_config(config, config_to_update_with)
+        assert config["api"][specific_key] == "PlaceHolder"
+        assert all(v != "PlaceHolder" for k, v in config["api"].items() if k != specific_key)
+
+
+value_mapping = [
+                ("SF_API_AUTHENTICATION_TOKEN","SomeAuth"),
+                ("SF_API_HOSTNAME","SomeHost"),
+                ("SF_API_USE_SSL","False"),
+                ("SF_API_PORT","56"),
+                ("SF_API_DEBUG","True")
+                ]
+
+parsed_values_mapping = {
+                "SF_API_AUTHENTICATION_TOKEN": "SomeAuth",
+                "SF_API_HOSTNAME": "SomeHost",
+                "SF_API_USE_SSL": False,
+                "SF_API_PORT": 56,
+                "SF_API_DEBUG": True,
+                        }
+
+class TestUpdateFromEnvironmentalVariables:
+    """Tests for the update_from_environment_variables function."""
+
+    def test_all_environment_variables_defined(self):
+        """Tests that the configuration object is updated correctly when all
+        the environment variables are defined."""
+
+        for key, value in value_mapping:
+            os.environ[key] = value
+
+        config = conf.create_config_object()
+        for v, parsed_value in zip(config["api"].values(), parsed_values_mapping.values()):
+            assert v != parsed_value
+
+        conf.update_from_environment_variables(config)
+        for v, parsed_value in zip(config["api"].values(), parsed_values_mapping.values()):
+            assert v == parsed_value
+
+        tear_down_all_env_var_defs()
+
+    environment_variables_with_keys_and_values = [
+                    ("SF_API_AUTHENTICATION_TOKEN","authentication_token","SomeAuth"),
+                    ("SF_API_HOSTNAME","hostname","SomeHost"),
+                    ("SF_API_USE_SSL","use_ssl","False"),
+                    ("SF_API_PORT","port", "56"),
+                    ("SF_API_DEBUG","debug","True")
+                    ]
+
+    @pytest.mark.parametrize("env_var, key, value", environment_variables_with_keys_and_values)
+    def test_one_environment_variable_defined(self, env_var, key, value):
+        """Tests that the configuration object is updated correctly when only
+        one environment variable is defined."""
+
+        tear_down_all_env_var_defs()
+        os.environ[env_var] = value
+
+        config = conf.create_config_object()
+        for v, parsed_value in zip(config["api"].values(), parsed_values_mapping.values()):
+            assert v != parsed_value
+
+        conf.update_from_environment_variables(config)
+        assert config["api"][key] == parsed_values_mapping[env_var]
+
+        for v, (key, parsed_value) in zip(config["api"].values(), parsed_values_mapping.items()):
+            if key != env_var:
+                assert v != parsed_value
+
+        # Tear-down
+        del os.environ[env_var]
```

Comment:
Yeah, so [apparently these are process specific changes](https://www.geeksforgeeks.org/python-os-environ-object/):

> `os.environ` behaves like a python dictionary, so all the common dictionary operations like get and set can be performed. We can also modify os.environ but any changes will be effective only for the current process where it was assigned and it will not change the value permanently.

The reason why they are included here is that if the entire test suite is run, then there might be state left between tests. That is, sometimes I came across funky situations :)

Reply:
Mentioned this earlier, but better to use `monkeypatch.setenv()` here to avoid side effects 🙂